pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/SegmentColumnarIndexCreator.java

```diff
@@ -363,7 +314,7 @@ private boolean createDictionaryForColumn(ColumnIndexCreationInfo info, SegmentG
 
   private boolean shouldCreateDictionaryWithinThreshold(ColumnIndexCreationInfo info,
       SegmentGeneratorConfig config, FieldSpec spec) {
-    long dictionarySize = info.getDistinctValueCount() * spec.getDataType().size();
+    long dictionarySize = info.getDistinctValueCount() * (long) spec.getDataType().size();
```

Comment:
nti: open a small PR to land this one-liner fix quickly?

Reply:
Added in https://github.com/apache/pinot/pull/10492